contracts/javascore/Intent_Contracts/app/src/main/java/network/icon/intent/Intent.java

```diff
@@ -0,0 +1,320 @@
+            BigInteger fee) {
+        Address toAddress = Address.fromString(_toAddress);
+        Address toTokenAddress = Address.fromString(_toToken);
+        if (toTokenAddress.equals(nativeAddress.get())) {
+            Context.require(Context.getValue().equals(amount.add(fee)), "\"Deposit amount not equal to order amount\"");
+            _nativeTransfer(toAddress, amount);
+            _nativeTransfer(feeHandler.get(), fee);
+        } else {
+            Context.call(toTokenAddress, "transferFrom", Context.getAddress(), toAddress,
+                    amount);
+            Context.call(toTokenAddress, "transferFrom", Context.getAddress(),
+                    feeHandler.get(), fee);
+        }
+    }
+
+    void _nativeTransfer(Address to, BigInteger amount) {
+        Context.transfer(to, amount);
+    }
+
+    public static byte[] hexStringToByteArray(String s) {
+        int len = s.length();
+        byte[] data = new byte[len / 2];
+        for (int i = 0; i < len; i += 2) {
+            data[i / 2] = (byte) ((Character.digit(s.charAt(i), 16) << 4)
+                    + Character.digit(s.charAt(i + 1), 16));
+        }
+        return data;
+    }
+
+    @External
+    public void tokenFallback(Address _from, BigInteger _value, byte[] _data) {
+        Context.require(_value.compareTo(BigInteger.ZERO) > 0, "Zero transfers not allowed");
+        String unpackedData = new String(_data);
+        Context.require(!unpackedData.equals(""), "Token Fallback: Data can't be empty");
+
+        JSONObject jsonObject = new JSONObject(unpackedData);
+        // string(address of depositer) -> string (deposits token address) -> amount of
+        // token
+        String depositor = jsonObject.get("depositor").toString();
+        String token = jsonObject.get("token").toString();
+        BigInteger amount = jsonObject.getBigInteger("amount");
+
+        deposit.at(depositor).set(token, amount);
+
+        SwapOrder swapOrder = SwapOrder
+                .fromBytes(hexStringToByteArray(jsonObject.get("swapOrderDataBytes").toString()));
+        Context.require(amount.equals(swapOrder.getAmount()), "Token amount must be equal");
+        Context.require(swapOrder.getToken() != null, "Token can't be null");
+        Context.require(Context.getCaller().toString().equals(swapOrder.getCreator()),
+                "Creator must be sender");
+
+        Address swapOrderToken = Address.fromString(swapOrder.getToken());
+        Address nativAddress = nativeAddress.get();
+        if (swapOrderToken.equals(nativAddress)) {
+            Context.require(Context.getValue().equals(swapOrder.getAmount()),
+                    "Deposit amount not equal to order amount");
+        } else {
+            Context.require(Context.getValue().equals(BigInteger.valueOf(0)),
+                    "Nativ Token Must Be Zero");
+            Context.call(swapOrderToken, "transfer", Context.getAddress(), swapOrder.getAmount());
```

Comment:
ICON tokens do not work like this, here you transfer to yourself and cause potentially a infinite loop